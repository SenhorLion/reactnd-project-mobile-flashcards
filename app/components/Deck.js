import React, { Component } from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';
import {
  black,
  purple,
  lightPurple,
  white,
  gray,
  antiFlashWhite,
} from '../utils/colors';
import { FontAwesome } from '@expo/vector-icons';
import ButtonTouchableOpacity from './ui/ButtonTouchableOpacity';

class Deck extends Component {
  render() {
    const {
      deck,
      deck: { id, title, questions },
      handleDeleteDeck,
      navigation,
    } = this.props;

    const hasDeck = !!(deck && Object.keys(deck).length);

    if (!hasDeck) {
      return (
        <View style={styles.deck}>
          <Text>No Deck exists</Text>
        </View>
      );
    }
    return (
      <View style={styles.deck}>
        <View style={styles.deckContainer}>
          <Text style={[styles.titleText, {}]}>{title}</Text>
          <Text style={styles.subText}>{`${questions.length} cards`}</Text>
          <ButtonTouchableOpacity
            backgroundColor={purple}
            marginTop={10}
            width={150}
            onPress={() => {
              return navigation.navigate('DeckDetail', {
                deck,
                deckId: id,
              });
            }}
          >
            <FontAwesome name="plus" size={20} color={antiFlashWhite} />
            <Text style={{ fontSize: 18, color: antiFlashWhite }}>
              View deck
            </Text>
          </ButtonTouchableOpacity>

          <ButtonTouchableOpacity
            backgroundColor={lightPurple}
            marginTop={10}
            width={150}
            onPress={() => handleDeleteDeck(deck)}
          >
            <FontAwesome name="trash" size={20} color={antiFlashWhite} />
            <Text style={{ fontSize: 18, color: antiFlashWhite }}>
              Delete deck
            </Text>
          </ButtonTouchableOpacity>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  deck: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: antiFlashWhite,
    marginLeft: 4,
    marginRight: 4,
    marginBottom: 4,
    minHeight: 180,
  },
  deckContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  titleText: {
    fontWeight: 'bold',
    textAlign: 'center',
    fontSize: 24,
    color: purple,
  },
  subText: {
    // flex: 1,
    textAlign: 'center',
    fontSize: 20,
    color: gray,
  },
});

export default Deck;
