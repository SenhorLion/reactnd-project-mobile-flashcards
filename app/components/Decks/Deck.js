import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { primary, antiFlashWhite } from '../../utils/colors';
import { FontAwesome } from '@expo/vector-icons';
import { ButtonTouchableOpacity } from '../Buttons';
import styles from './styles';
import { pluralize } from '../../utils/helpers';

class Deck extends Component {
  render() {
    const { deck, deck: { id, title, questions }, navigation } = this.props;
    const numOfCards = questions.length;

    return (
      <View style={styles.deck}>
        <View style={styles.deckContainer}>
          <Text style={styles.titleText}>{title}</Text>
          <Text style={styles.subText}>{`${numOfCards} ${pluralize(
            'card',
            numOfCards
          )}`}</Text>
          <ButtonTouchableOpacity
            backgroundColor={primary}
            marginTop={10}
            width={150}
            onPress={() => {
              return navigation.navigate('DeckDetail', {
                deck,
                deckId: id,
              });
            }}
          >
            <FontAwesome name="plus" size={20} color={antiFlashWhite} />
            <Text style={{ fontSize: 18, color: antiFlashWhite }}>
              View deck
            </Text>
          </ButtonTouchableOpacity>
        </View>
      </View>
    );
  }
}

export default Deck;
