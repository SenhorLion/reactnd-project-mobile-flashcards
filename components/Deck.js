import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';
import {
  black,
  purple,
  lightPurple,
  white,
  gray,
  antiFlashWhite,
} from '../utils/colors';
import { FontAwesome } from '@expo/vector-icons';

const Deck = ({ deck, deck: { title, questions }, navigation }) => {
  const hasDeck = !!(deck && Object.keys(deck).length);

  if (!hasDeck) {
    return (
      <View style={styles.deck}>
        <Text>No Deck exists</Text>
      </View>
    );
  }
  return (
    <View style={styles.deck}>
      <View style={styles.deckContainer}>
        <Text style={[styles.titleText, {}]}>{title}</Text>
        <Text style={styles.subText}>{`${questions.length} cards`}</Text>
        <TouchableOpacity
          style={[styles.button, { marginTop: 20, width: 150 }]}
          onPress={() => {
            return navigation.navigate('DeckDetail', { deck, entryId: title });
          }}
        >
          <FontAwesome name="plus" size={20} color={antiFlashWhite} />
          <Text style={{ fontSize: 18, color: antiFlashWhite }}>View deck</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  deck: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: antiFlashWhite,
    marginLeft: 4,
    marginRight: 4,
    marginBottom: 4,
    minHeight: 180,
  },
  deckContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  titleText: {
    fontWeight: 'bold',
    textAlign: 'center',
    fontSize: 24,
    color: purple,
  },
  subText: {
    // flex: 1,
    textAlign: 'center',
    fontSize: 20,
    color: gray,
  },
  button: {
    flexDirection: 'row',
    padding: 4,
    margin: 4,
    height: 40,
    width: 80,
    alignItems: 'center',
    justifyContent: 'space-around',
    backgroundColor: purple,
  },
});

export default Deck;
