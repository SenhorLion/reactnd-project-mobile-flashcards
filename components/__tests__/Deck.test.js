import React from 'react';
import TestRenderer from 'react-test-renderer';
import Deck from '../Deck';

it('renders Deck component without crashing', () => {
  const decksMock = {
    React: {
      title: 'React',
      questions: [
        {
          question: 'What is React?',
          answer: 'A library for managing user interfaces',
        },
        {
          question: 'Where do you make Ajax requests in React?',
          answer: 'The componentDidMount lifecycle event',
        },
      ],
    },
  };
  const DeckComponent = TestRenderer.create(<Deck deck={decksMock} />).toJSON();

  expect(DeckComponent).toMatchSnapshot();
});

it.only('renders no deck avaialble message when no deck exists', () => {
  const decksMock = {};
  const DeckComponent = TestRenderer.create(<Deck deck={decksMock} />).toJSON();

  expect(DeckComponent).toMatchSnapshot();
});
