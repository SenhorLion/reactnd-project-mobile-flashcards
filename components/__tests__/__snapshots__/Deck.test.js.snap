// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Deck component without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f2f3f4",
      "flex": 1,
      "justifyContent": "center",
      "marginBottom": 4,
      "marginLeft": 4,
      "marginRight": 4,
      "minHeight": 180,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#292477",
            "fontSize": 24,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      React
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#757575",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      2 cards
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#292477",
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "space-around",
          "margin": 4,
          "marginTop": 20,
          "opacity": 1,
          "padding": 4,
          "width": 150,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#f2f3f4",
            "fontSize": 18,
          }
        }
      >
        View deck
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#7c53c3",
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "space-around",
          "margin": 4,
          "marginTop": 10,
          "opacity": 1,
          "padding": 4,
          "width": 150,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#f2f3f4",
            "fontSize": 18,
          }
        }
      >
        Delete deck
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders deck 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f2f3f4",
      "flex": 1,
      "justifyContent": "center",
      "marginBottom": 4,
      "marginLeft": 4,
      "marginRight": 4,
      "minHeight": 180,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#292477",
            "fontSize": 24,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      React
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#757575",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      2 cards
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#292477",
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "space-around",
          "margin": 4,
          "marginTop": 20,
          "opacity": 1,
          "padding": 4,
          "width": 150,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#f2f3f4",
            "fontSize": 18,
          }
        }
      >
        View deck
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#7c53c3",
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "space-around",
          "margin": 4,
          "marginTop": 10,
          "opacity": 1,
          "padding": 4,
          "width": 150,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#f2f3f4",
            "fontSize": 18,
          }
        }
      >
        Delete deck
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders no deck avaialble message when no deck exists 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f2f3f4",
      "flex": 1,
      "justifyContent": "center",
      "marginBottom": 4,
      "marginLeft": 4,
      "marginRight": 4,
      "minHeight": 180,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    No Deck exists
  </Text>
</View>
`;
